Magento/Sniffs/Exceptions/LocalizedThrowSniff.php

```diff
@@ -0,0 +1,78 @@
+<?php
+/**
+ * Copyright © Magento. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+
+namespace Magento\Sniffs\Exceptions;
+
+use PHP_CodeSniffer\Sniffs\Sniff;
+use PHP_CodeSniffer\Files\File;
+
+/**
+ * Detects possible direct throws of Exceptions.
+ */
+class LocalizedThrowSniff implements Sniff
+{
+
+    protected $presentationLayerKeys = ['Controller', 'Block', 'ViewModel'];
+
+    /**
+     * String representation of warning.
+     */
+    protected $warningMessage = 'LocalizedException SHOULD only be thrown in the Presentation layer.';
```

Comment:
It looks like tech guidelines must be updated to allow LocalizedException in Service Layer as well. See #5.17 in https://devdocs.magento.com/guides/v2.3/coding-standards/technical-guidelines.html

Please do not merge this PR until tech guidelines are updated.